requirements/specifications/openrpc-extensions/app-passthrough-apis.md

```diff
@@ -0,0 +1,533 @@
+
+Example platform method with context:
+```json
+{
+    "name": "onFoo",
+    "tags": [
+        {
+            "name": "capabilities",
+            "x-uses": [
+                "xrn:firebolt:capabilities:example:foo"
+            ],
+            "x-provided-by": "foo"
+        },
+        {
+            "name": "event"
+        }
+    ],
+    "params": [
+        {
+            "name": "context1",
+            "schema":{
+                "type": "string"
+            }
+        },
+        {
+            "name": "context2",
+            "schema": {
+                "type": "number"
+            }
+        }
+    ],
+    "result": {
+        "name": "value",
+        "schema": {
+            "type": "boolean"
+        }
+    }
+}
+```
+
+Matching provider method:
+
+```json
+{
+    "name": "foo",
+    "tags": [
+        {
+            "name": "capabilities",
+            "x-provides": "xrn:firebolt:capabilities:example:foo"
+        },
+        {
+            "name": "event"
+        }
+    ],
+    "params": [
+        {
+            "name": "context1",
+            "schema":{
+                "type": "string"
+            }
+        },
+        {
+            "name": "context2",
+            "schema": {
+                "type": "number"
+            }
+        },
+        {
+            "name": "value",
+            "schema": {
+                "type": "boolean"
+            }
+        }
+    ]
+}
+```
+
+When a provider app calls a provider method mapped to an event the platform **MUST** ignore the notification if the app is not a [candidate app](#7-provider-candidates) for this capability.
+
+If the platform method result schema matches the *last* parameter schema on the provider method then the value **MUST** be used as-is.
+
+Otherwise if the platform method result schema has a property that matches the *last* parameter schema on the provider method then the value **MUST** be composed into an object under the corresponding property name and the platform **MUST** apply any [result transformations](#9-result-transformations) to the composed result.
+
+Finally the platform **MUST** dispatch the notification to the app that registered for the event via the original platform method, using all but the last parameter as context.
+
+## 5. Provider Candidates
+The Firebolt Device Manfist **MUST** have a list of `ProviderPolicy` configurations that map capabilities to policies for determining candidate providers:
+
+```json
+{
+    "providerPolicies": [
+        {
+            "inFocus": true,
+            "capabilities": [
+                "xrn:firebolt:capability:foo:bar"
+            ]
+        }
+    ]
+}
+```
+The policy **MUST** have a list of capabilities that it is applied to.
+
+A capability **MUST NOT** be included in more than one policy.
+
+The policy **MAY** have an `inFocus` boolean.
+
+If the policy has `inFocus` set to `true` then any app without RCU input focus when the capability is invoked **MUST NOT** be considered a candidate.
```

Comment:
is inFocus=false allowed? Apps with RCU input focus are NOT considered?

Reply:
hmmm, I was thinking that `inFocus: false` would be the same as leaving it out, so it means we don't care if the app is in focus or not.